src/front.coffee

```diff
@@ -1,28 +1,76 @@
+moment = require 'moment'
+AbstractAPIAdapter = require 'hubot-abstract-api-adapter'
 try
-	{ Robot, Adapter, TextMessage, User } = require 'hubot'
+	{ TextMessage } = require 'hubot'
 catch
 	prequire = require 'parent-require'
-	{ Robot, Adapter, TextMessage, User } = prequire 'hubot'
-
-class Front extends Adapter
+	{ TextMessage } = prequire 'hubot'
 
+class Front extends AbstractAPIAdapter
 	constructor: ->
-		super
-		@robot.logger.info 'Constructor'
+		try
+			super
+		catch error
+			if error not instanceof TypeError
+				throw error
+
+	extractNext: (obj) -> obj._pagination?.next
+
+	extractResults: (obj) -> obj._results
 
-	send: (envelope, strings...) ->
-		@robot.logger.info 'Send'
+	poll: =>
+		checkHistoryUntil = @robot.brain.get 'AdapterFrontLastKnown'
+		if not checkHistoryUntil?
+			firstCheckWindow = parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY ? '24')
+			checkHistoryUntil = moment().subtract(firstCheckWindow, 'h').unix()
+		@getUntil(
+			@getOptions process.env.HUBOT_FRONT_API_URL + '/events'
+			@processMessage
+			(obj) -> obj.emitted_at > checkHistoryUntil
+			(err, res) => if not err then @robot.brain.set('AdapterFrontLastKnown', res[0].emitted_at)
+		)
 
-	reply: (envelope, strings...) ->
-		@robot.logger.info 'Reply'
+	getOptions: (url) ->
+		{
+			url: url
+			headers: {
+				Accept: 'application/json'
+				Authorization: 'Bearer ' + process.env.HUBOT_FRONT_API_TOKEN
+			}
+			qs: {
+				before: Date.now()
+			}
+		}
```

Comment:
These braces aren't necessary, although I guess it comes down to personal preference.